contribs/src/main/scala/io/delta/storage/DynamoDBLogStore.scala

```diff
@@ -0,0 +1,209 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.storage
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.dynamodbv2.model.{
+  AttributeValue,
+  ComparisonOperator,
+  Condition,
+  ConditionalCheckFailedException,
+  ExpectedAttributeValue,
+  GetItemRequest,
+  PutItemRequest,
+  QueryRequest
+}
+import java.util.NoSuchElementException
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.SparkConf
+import com.amazonaws.regions.Region
+import com.amazonaws.regions.Regions
+
+/*
+  DynamoDB requirements:
+
+  Single dynamodb table is required. Default table name is 'delta_log',
+  it may be changed by setting spark property.
+
+  Required key schema:
+   - tablePath: String, HASH
+   - filename: String, RANGE
+
+  Table may be created with following aws cli command:
+
+    aws --region us-east-1 dynamodb create-table \
+      --table-name delta_log \
+      --attribute-definitions AttributeName=tablePath,AttributeType=S \
+                              AttributeName=filename,AttributeType=S \
+      --key-schema AttributeName=tablePath,KeyType=HASH \
+                   AttributeName=filename,KeyType=RANGE \
+      --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5
+
+  Following spark properties are recognized:
+  - spark.delta.DynamoDBLogStore.tableName - table name (defaults to 'delta_log')
+  - spark.delta.DynamoDBLogStore.endpoint - endpoint (defaults to 'Amazon AWS')
+  - spark.delta.DynamoDBLogStore.region - AWS region (defaults to 'us-east-1')
+  - spark.delta.DynamoDBLogStore.credentials.provider - name of class implementing
+    `com.amazonaws.auth.AWSCredentialsProvider` (defaults to 'DefaultAWSCredentialsProviderChain')
+ */
+class DynamoDBLogStore(sparkConf: SparkConf, hadoopConf: Configuration)
+    extends BaseExternalLogStore(sparkConf, hadoopConf) {
+
+  import DynamoDBLogStore._
+
+  private val tableName =
+    sparkConf.get(s"${confPrefix}tableName", "delta_log")
+
+  private val client: AmazonDynamoDBClient =
+    DynamoDBLogStore.getClient(sparkConf)
+
+  override protected def putDbEntry(
+      entry: LogEntry,
+      overwrite: Boolean = false
+  ): Unit = {
+    try {
+      logDebug(s"putItem $entry, overwrite: $overwrite")
+      client.putItem(entry.asPutItemRequest(tableName, overwrite))
+    } catch {
+      case e: ConditionalCheckFailedException =>
+        logDebug(e.toString)
+        throw new java.nio.file.FileAlreadyExistsException(
+          entry.absoluteJsonPath.toString
+        )
+    }
+  }
+
+  override protected def getDbEntry(
```

Comment:
document that `absoluteTablePath` --> partition key, and that `absoluteJsonPath.getName` --> sort key